src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -173,42 +189,93 @@ def ev_charger_pool(
     def battery_pool(
         self,
         battery_ids: abc.Set[int] | None = None,
+        name: str | None = None,
+        priority: int = -sys.maxsize - 1,
```

Comment:
Any reason not to make the `priority` a `float`, so we don't end up writing *BASIC* to make sure we can sneak in some new priority between 2 existing priorities (priority 10, 20, 30, 40.... :D).

Also I don´t get the default. Why not 0?

Reply:
Considering that:

* We only care about ordering
* If 2 priorities are conceptually equal, we don't care which one is considered higher than the other (in cases where `float` could not compare equal for 2 equal numbers, like `assert 0.1 + 0.2 > 0.3`)
* We'll probably be using small (but not too small) literals
 
It should be reasonably safe to use `float`s IMHO.

The problem with integers is if we end up with 2 actors using consecutive numbers there is no (external) way to sneak something between both, and you might not have control over those 2 actors to go change the code and redeploy. I we would want to be extra safe, we could add a `Priority(float)` object that checks if the `float` is perfectly representable and raises if it's not, but as a first step I would use `float`.

Reply:
BTW, I'm fine with merging as is and discuss this separate and do a follow-up PR if we decide to change it.

Reply:
Also, I just saw the commit that makes `-sys.maxsize - 1` the default. I'm not sure why we want to have the lowest possible priority ever the default, that is not explained in the commit message.

Reply:
**int vs float:**

My feeling has always been that priorities have to be externally controlled, possibly through UI config.  Because actor implementations by themselves can't know what their priorities are.

But going from int to float is not a discussion, so I agree that we can discuss separately.

**default priority:**

This is of course subjective, and is based on some further opinions: 
- The default priority is used only when an actor knows it will not be sharing batteries with other actors.  
- Most actors don't know this, so they should provide a way to configure their priorities.  
- If an actor uses the default priority (i.e doesn't care about its priority), it should be lower than any other actor, even the ones with negative priorities, because negative priorities doesn't mean that they're low priority, just that the priority numbers are negative.  
- If actors care about priorities, they should provide a way to specify one, else they get put in the bottom of the pile.

Reply:
Yeah, let's leave it for when we have a better idea of how this will be used and who will be setting priorities and how.